custom/icds_reports/utils/data_pull_ap_rj.sql

```diff
@@ -0,0 +1,271 @@
+COPY(SELECT
+    state_name,
+    district_name,
+    block_name,
+    supervisor_name,
+    awc_name,
+    awc_site_code,
+    CASE WHEN SUM(CASE WHEN month='2019-04-01'  THEN thr_eligible ELSE 0 END)>0 THEN round((sum(CASE WHEN num_rations_distributed>=21 AND month='2019-04-01'  THEN 1 ELSE 0 END)/SUM(CASE WHEN month='2019-04-01'  THEN thr_eligible ELSE 0 END)::float)::numeric,2)*100 ELSE 0 END as pw_thr_21_April,
+    CASE WHEN SUM(CASE WHEN month='2019-05-01'  THEN thr_eligible ELSE 0 END)>0 THEN round((sum(CASE WHEN num_rations_distributed>=21 AND month='2019-05-01'  THEN 1 ELSE 0 END)/SUM(CASE WHEN month='2019-05-01'  THEN thr_eligible ELSE 0 END)::float)::numeric,2)*100 ELSE 0  END as pw_thr_21_May,
+    CASE WHEN SUM(CASE WHEN month='2019-06-01'  THEN thr_eligible ELSE 0 END)>0 THEN round((sum(CASE WHEN num_rations_distributed>=21 AND month='2019-06-01'  THEN 1 ELSE 0 END)/SUM(CASE WHEN month='2019-06-01'  THEN thr_eligible ELSE 0 END)::float)::numeric,2)*100 ELSE 0  END as pw_thr_21_June,
+    CASE WHEN SUM(CASE WHEN month='2019-07-01'  THEN thr_eligible ELSE 0 END)>0 THEN round((sum(CASE WHEN num_rations_distributed>=21 AND month='2019-07-01'  THEN 1 ELSE 0 END)/SUM(CASE WHEN month='2019-07-01'  THEN thr_eligible ELSE 0 END)::float)::numeric,2)*100 ELSE 0  END as pw_thr_21_July,
+    CASE WHEN SUM(CASE WHEN month='2019-08-01'  THEN thr_eligible ELSE 0 END)>0 THEN round((sum(CASE WHEN num_rations_distributed>=21 AND month='2019-08-01'  THEN 1 ELSE 0 END)/SUM(CASE WHEN month='2019-08-01'  THEN thr_eligible ELSE 0 END)::float)::numeric,2)*100 ELSE 0  END as pw_thr_21_Aug,
+    CASE WHEN SUM(CASE WHEN month='2019-09-01'  THEN thr_eligible ELSE 0 END)>0 THEN round((sum(CASE WHEN num_rations_distributed>=21 AND month='2019-09-01'  THEN 1 ELSE 0 END)/SUM(CASE WHEN month='2019-09-01'  THEN thr_eligible ELSE 0 END)::float)::numeric,2)*100 ELSE 0  END as pw_thr_21_Sept,
+    CASE WHEN SUM(CASE WHEN month='2019-10-01'  THEN thr_eligible ELSE 0 END)>0 THEN round((sum(CASE WHEN num_rations_distributed>=21 AND month='2019-10-01'  THEN 1 ELSE 0 END)/SUM(CASE WHEN month='2019-10-01'  THEN thr_eligible ELSE 0 END)::float)::numeric,2)*100 ELSE 0  END as pw_thr_21_Oct,
+    CASE WHEN SUM(CASE WHEN month='2019-11-01'  THEN thr_eligible ELSE 0 END)>0 THEN round((sum(CASE WHEN num_rations_distributed>=21 AND month='2019-11-01'  THEN 1 ELSE 0 END)/SUM(CASE WHEN month='2019-11-01'  THEN thr_eligible ELSE 0 END)::float)::numeric,2)*100 ELSE 0  END as pw_thr_21_Nov
+FROM
+    awc_location  LEFT JOIN  "ccs_record_monthly"  ON (
+        ccs_record_monthly.supervisor_id=awc_location.supervisor_id and
+        ccs_record_monthly.awc_id = awc_location.doc_id AND
+        ccs_record_monthly.pregnant=1
+    )
+    where month in ('2019-04-01', '2019-05-01', '2019-06-01', '2019-07-01','2019-08-01','2019-09-01','2019-10-01', '2019-11-01')
+and   awc_location.state_id='f98e91aa003accb7b849a0f18ebd7039' and awc_location.aggregation_level=5
+GROUP by state_name,
```

Comment:
grouping by supervisor_id as well should allow it to do grouping on the workers rather than the master